bin/post-process-sce.R

```diff
@@ -0,0 +1,150 @@
+    opt_str = c("-o", "--output_sce_file"),
+    type = "character",
+    help = "path to output rds file to store processed sce object. Must end in .rds"
+  ),
+  make_option(
+    opt_str = c("--prob_compromised_cutoff"),
+    type = "double",
+    default = 0.75,
+    help = "probability compromised cutoff used for filtering cells with miQC",
+    metavar = "double"
+  ),
+  make_option(
+    opt_str = c("--gene_cutoff"),
+    type = "integer",
+    default = 200,
+    help = "Minimum number of genes per cell cutoff used for filtering cells."
+  ),
+  make_option(
+    opt_str = c("--n_hvg"),
+    type = "integer",
+    default = 2000,
+    help = "number of high variance genes to use for dimension reduction;
+            the default is n_hvg = 2000",
+  ),
+  make_option(
+    opt_str = c("--n_pcs"),
+    type = "integer",
+    default = 50,
+    help = "Number of principal components to retain in the returned SingleCellExperiment object 
+      when using scater::runPCA; the default is n_pcs = 50",
+  ),
+  make_option(
+    opt_str = c("-r", "--random_seed"),
+    type = "integer",
+    help = "A random seed for reproducibility."
+  )
+)
+
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# set seed
+set.seed(opt$random_seed)
+
+# check that input file file exists
+if(!file.exists(opt$input_sce_file)){
+  stop("Missing unfiltered.rds file")
+}
+
+# check that output file name ends in .rds
+if(!(stringr::str_ends(opt$output_sce_file, ".rds"))){
+  stop("output file name must end in .rds")
+}
+
+# read in input rds file
+sce <- readr::read_rds(opt$input_sce_file)
+
+# check that prob compromised cutoff is between 0-1
+if(!dplyr::between(opt$prob_compromised_cutoff, 0, 1)){
+  stop("--prob_compromised_cutoff must be a number between 0 to 1")
+}
+
+# create ccdl_filter column
+if(all(is.na(sce$prob_compromised))){
+  colData(sce)$ccdl_filter <- ifelse(
+    sce$detected >= opt$gene_cutoff, "Keep", "Remove"
+  )
+  metadata(sce)$ccdl_filter_method <- "Minimum_gene_cutoff"
+} else {
+  # remove cells with >= probability compromised cutoff + min gene cutoff
+  colData(sce)$ccdl_filter <- ifelse(
+    sce$prob_compromised < opt$prob_compromised_cutoff & 
+      sce$detected >= opt$gene_cutoff, 
+    "Keep", 
+    "Remove"
+  )
+  metadata(sce)$ccdl_filter_method <- "miQC"
+}
+
+# filter sce using criteria in ccdl_filter 
+filtered_sce <- sce[, sce$ccdl_filter == "Keep"]
+
+# cluster prior to normalization 
+qclust <- NULL
+tryCatch({
+  # Cluster similar cells
+  qclust <- scran::quickCluster(filtered_sce) 
+  
+  # Compute sum factors for each cell cluster grouping
+  filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
+  
+  # Include note in metadata re: clustering before computing sum factors
+  metadata(filtered_sce)$normalization <- "deconvolution"
+})
+
+if (is.null(qclust)) {
+  
+  # Include note in metadata re: failed clustering
+  metadata(filtered_sce)$normalization <- "log-normalization"
+  
```

Comment:
Just removing some excess whitespace.
```suggestion
  # Include note in metadata re: failed clustering
  metadata(filtered_sce)$normalization <- "log-normalization"
```